deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,471 @@
+@pytest.fixture(autouse=True, scope="module")
+def remove_the_database_file_after_compaction_session_tests_complete():
+    if os.path.exists(DATABASE_FILE_PATH_VALUE):
+        os.remove(DATABASE_FILE_PATH_VALUE)
+
+
+"""
+FUNCTION scoped fixtures
+"""
+
+
+# SETUP
+@pytest.fixture(scope="function")
+def ds_mock_kwargs():
+    # see deltacat/tests/local_deltacat_storage/README.md for documentation
+    kwargs_for_local_deltacat_storage: Dict[str, Any] = {
+        DATABASE_FILE_PATH_KEY: DATABASE_FILE_PATH_VALUE,
+    }
+    yield kwargs_for_local_deltacat_storage
+    if os.path.exists(DATABASE_FILE_PATH_VALUE):
+        os.remove(DATABASE_FILE_PATH_VALUE)
+
+
+# TEARDOWN
+@pytest.fixture(scope="function")
+def cleanup_database_between_executions():
+    if os.path.exists(DATABASE_FILE_PATH_VALUE):
+        os.remove(DATABASE_FILE_PATH_VALUE)
+
+
+INCREMENTAL_TEST_CASES = {
+    "1-incremental-pkstr-sknone-norcf": (
+        {"pk_col_1"},  # Primary key columns
+        [],  # Sort key columns
+        [{"key_name": "region_id", "key_type": "int"}],  # Partition keys
+        ["pk_col_1"],  # column_names
+        [pa.array([str(i) for i in range(10)])],  # arrow arrays
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [pa.array([str(i) for i in range(10)])], names=["pk_col_1"]
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "2-incremental-pkstr-skstr-norcf": (
+        ["pk_col_1"],
+        [
+            {
+                "key_name": "sk_col_1",
+            }
+        ],
+        [],
+        ["pk_col_1", "sk_col_1"],
+        [pa.array([str(i) for i in range(10)]), pa.array(["test"] * 10)],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [pa.array([str(i) for i in range(10)]), pa.array(["test"] * 10)],
+            names=["pk_col_1", "sk_col_1"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "3-incremental-pkstr-multiskstr-norcf": (
+        ["pk_col_1"],
+        [
+            {
+                "key_name": "sk_col_1",
+            },
+            {
+                "key_name": "sk_col_2",
+            },
+        ],
+        [],
+        ["pk_col_1", "sk_col_1", "sk_col_2"],
+        [
+            pa.array([str(i) for i in range(10)]),
+            pa.array(["test"] * 10),
+            pa.array(["foo"] * 10),
+        ],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [
+                pa.array([str(i) for i in range(10)]),
+                pa.array(["test"] * 10),
+                pa.array(["foo"] * 10),
+            ],
+            names=["pk_col_1", "sk_col_1", "sk_col_2"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "4-incremental-duplicate-pk": (
```

Comment:
Please add few more test cases for duplicate pk with decimal columns, and int columns and multi pk as well.